src/components/avatar/avatar.ts

```diff
@@ -0,0 +1,47 @@
+import { Block } from '~src/utils/block';
+import { Router } from '~src/utils/router';
+import { API_HOST } from '~src/utils/constants';
+import avatarTemplate from './avatar.tmpl.pug';
+import unionImage from '../../../static/images/union.svg';
+import { connect } from '~src/utils/connect';
+
+const withAvatar = connect((state) => ({
+    avatar: state.user?.avatar
+        ? `${API_HOST}/resources${state.user.avatar}`
+        : '',
+}));
+
+class Avatar extends Block {
+    router: Router;
+
+    constructor(props) {
+        super('div', props);
+
+        this.router = new Router('');
+    }
+
+    protected getAttributes(): Record<string, string> {
+        return {
+            class: 'profile-avatar',
+        };
+    }
+
+    protected getEvents(): Record<string, (e: Event) => void> {
+        return {
+            click: (event) => {
+                event.preventDefault();
+
+                if (this.props?.changeHref)
+                    this.router.go(this.props.changeHref);
```

Comment:
По конвенции условия лучше писать полностью
```suggestion
                if (this.props?.changeHref) {
                    this.router.go(this.props.changeHref);
                }
```

Reply:
не понял. можно пример?